drivers/eeprom/include/eeprom_driver.h

```diff
@@ -0,0 +1,150 @@
+ * @ref eeprom_driver_t::erase can be `NULL`.
+ */
+typedef struct {
+    /**
+     * @brief   Initialize the data structures for the EEPROM device
+     *
+     * @param[in,out]   handle  Device handle to initialize
+     * @param[in]       params  Initialization parameters of the device
+     *
+     * @retval  -1      Failed to initialize the device
+     * @return          The size of the device in bytes
+     */
+    ssize_t (*init)(void *handle, const void *params);
+    /**
+     * @brief   Read data from the EEPROM device
+     *
+     * @param[in,out]   handle  Device handle of the EEPROM to read from
+     * @param[in]       pos     Position to start to read from (in bytes)
+     * @param[out]      dest    Buffer to write the data to
+     * @param[in]       len     Number of bytes to read
+     *
+     * @retval  0       Success
+     * @retval  -EIO    I/O error occurred while reading
+     */
+    int (*read)(void *handle, size_t pos, void *dest, size_t len);
+    /**
+     * @brief   Write data from the EEPROM device
+     *
+     * @param[in,out]   handle  Device handle of the EEPROM to write to
+     * @param[in]       pos     Position to start to write to (in bytes)
+     * @param[in]       data    Buffer to write to the EEPROM device
+     * @param[in,out]   len     Number of bytes to write
+     *
+     * @retval  0       Success
+     * @retval  -EIO    I/O error occurred while writen
+     *
+     * If `-EFAULT` has been returned, the content of the EEPROM device is
+     * unchanged. If `0` is returned, all data has been written successfully
+     * to the device. If `-EIO` is returned, the contents in `[pos; pos + len[`
+     * of the device are undefined, but the remaining content of the device
+     * is unchanged.
+     */
+    int (*write)(void *handle, size_t pos, const void *data, size_t len);
+    /**
+     * @brief   Set all bytes in `[pos; pos + len[` to the value in @p val
+     *
+     * @param[in,out]   handle  Device handle of the EEPROM to write to
+     * @param[in]       pos     Position to start to write to (in bytes)
+     * @param[in]       val     Write this in every byte in `[pos; pos + len[`
+     * @param[in        len     Number of bytes to set to @p val
+     *
+     * @retval  0       Success
+     * @retval  -EIO    I/O error occurred while writen
+     *
+     * If `-EFAULT` has been returned, the content of the EEPROM device is
+     * unchanged. If `0` is returned, all data has been written successfully
+     * to the device. If `-EIO` is returned, the contents in `[pos; pos + len[`
+     * of the device are undefined, but the remaining content of the device
+     * is unchanged.
+     *
+     * This function is optional to implement by the driver. If the function
+     * pointer is `NULL`, a fallback implementation that calls
+     * @ref eeprom_driver_t::write is used instead.
+     */
+    int (*set)(void *handle, size_t pos, uint8_t val, size_t len);
```

Comment:
set and erase function are not strictly needed since they just require the write method. Maybe we could simply the driver struct with just init/read/write functions ?

Reply:
Some EEPROMs do offer opcodes to do page-wise erase, might me faster than manually overwriting all the bytes.

Reply:
That's why the `eeprom_driver_t` has an `erase()` function pointer. Only if that is `NULL`, the fallback implementation of using `set()` is used.

(The global `erase()` function is intended to be used by the user to erase memory. Maybe extending the doc to point to that would be a good idea.)

Reply:
ok for erase but this is not the case for set since it can be implemented using the write function (like clear).

Reply:
done